### PR TITLE
Sets hostname to fdqn when dns_domain is available

### PR DESCRIPTION
**Pull request by lorengordon**

### CODE REVIEW

**Comment by lorengordon** [comment]
Build showing this isn't breaking anything:

https://us-east-1.console.aws.amazon.com/codesuite/codebuild/701759196663/projects/terrafirm/build/terrafirm%3Ab7f2bb7f-c535-429f-bb22-0da814e58ade?region=us-east-1

File: appveyor.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ environment:
   global:
     SALT_FILEROOT: '%APPVEYOR_BUILD_FOLDER%'
     SALT_STATE: name-computer
-    SALT_URL: 'https://repo.saltstack.com/windows/Salt-Minion-2018.3.4-Py2-AMD64-Setup.exe'
+    SALT_URL: https://repo.saltproject.io/windows/Salt-Minion-3004.2-1-Py3-AMD64-Setup.exe
```
Comment (ferricoxide) [inline appveyor.yml:7]: So, when we move to 3005.1+ (so that this formula works for EL8), this value will need to be updated?

File: .travis.yml [hunk 2]
```diff
@@ -1,19 +1,26 @@
...
+        - SALT_GRAIN_VALUE=False
+        - SALT_REPO_URL=https://repo.saltproject.io/py3/redhat/7/x86_64/3004.repo
+      services:
+        - docker
+    - env:
+        - OS_VERSION=7
+        - SALT_GRAIN_VALUE=itworx
+        - SALT_REPO_URL=https://repo.saltproject.io/py3/redhat/7/x86_64/3004.repo
+      services:
+        - docker
```
Comment (ferricoxide) [inline .travis.yml:23]: Presumably, when we moved to 3005.1+ (for EL8 compatibility), we'll need to add testing-supporting values for EL8?

**Review by ferricoxide — APPROVED** [review]
Approved - just had questions about inbound change-requirements.

File: appveyor.yml (hunk 1 above)
Comment (lorengordon) [reply appveyor.yml:7]: Indeed. We don't yet have a way of letting dependabot handle these version updates for us.

**Pull request merged**